### PR TITLE
feat(lambda): W3C trace context support for Lambda tracing

### PR DESCRIPTION
**Pull request by basti1302**

### CODE REVIEW

File: packages/aws-lambda/test/spec_compliance/lambda.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+// assumed root dir of the Lambda.
+require('../../src/metrics/rootDir').root = require('path').resolve(__dirname, '..', '..');
+
+const fetch = require('node-fetch');
+
+const downstreamDummyUrl = process.env.DOWNSTREAM_DUMMY_URL;
+
+exports.handler = instana.wrap(async event => {
+  const { headers, httpMethod, path } = event;
+  console.log(`-> ${httpMethod} ${path} ${JSON.stringify(headers)}`);
```
Comment (willianpc) [inline packages/aws-lambda/test/spec_compliance/lambda.js:24]: are these `console.log`s intended to stay here?

File: packages/aws-lambda/test/spec_compliance/lambda.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  const downstreamResponse = await fetch(downstreamDummyUrl);
+  const downstreamResponseBody = await downstreamResponse.json();
```
Comment (willianpc) [inline packages/aws-lambda/test/spec_compliance/lambda.js:26]: Is there any chance that one of these `await` statements could be rejected?

**Review by willianpc — APPROVED** [review]
LGTM.
Added two questions out of curiosity. Unless the await statement can be rejected, which in this case you may consider adding a try/catch around them :)

File: packages/aws-lambda/test/spec_compliance/lambda.js (hunk 2 above)
Comment (basti1302) [reply packages/aws-lambda/test/spec_compliance/lambda.js:24]: They should not fail but if they do, packages/aws-lambda/test/runtime_mock#runHandler will catch that error and the test will fail with a nice, clean error message.

File: packages/aws-lambda/test/spec_compliance/lambda.js (hunk 1 above)
Comment (basti1302) [reply packages/aws-lambda/test/spec_compliance/lambda.js]: Removed them.

**Pull request merged**